scripts/dev/e2e.py

```diff
@@ -129,30 +108,89 @@ def build_and_push_version_upgrade_hook(repo_url: str, tag: str, path: str) -> N
     build_and_push_image(repo_url, tag, path, "versionhook")
 
 
-def _delete_testrunner_pod(config_file: str) -> None:
+def _delete_test_pod(config_file: str) -> None:
     """
     _delete_testrunner_pod deletes the test runner pod
     if it already exists.
     """
     dev_config = load_config(config_file)
     corev1 = client.CoreV1Api()
     k8s_conditions.ignore_if_doesnt_exist(
-        lambda: corev1.delete_namespaced_pod(TEST_RUNNER_NAME, dev_config.namespace)
+        lambda: corev1.delete_namespaced_pod(TEST_POD_NAME, dev_config.namespace)
     )
 
 
-def create_test_runner_pod(args: argparse.Namespace) -> None:
-    """
-    create_test_runner_pod creates the pod which will run all of the tests.
-    """
-    dev_config = load_config(args.config_file)
+def create_test_pod(args: argparse.Namespace, dev_config: DevConfig) -> None:
     corev1 = client.CoreV1Api()
-    pod_body = _get_testrunner_pod_body(args)
-
+    test_pod = {
+        "kind": "Pod",
+        "metadata": {
+            "name": TEST_POD_NAME,
+            "namespace": dev_config.namespace,
+        },
+        "spec": {
+            "restartPolicy": "Never",
+            "serviceAccountName": "mongodb-kubernetes-operator",
+            "containers": [
+                {
+                    "name": TEST_POD_NAME,
+                    "image": f"{dev_config.repo_url}/{dev_config.e2e_image}:{args.tag}",
+                    "imagePullPolicy": "Always",
+                    "volumeMounts": [
+                        {"mountPath": "/etc/config", "name": "kube-config-volume"}
+                    ],
+                    "env": [
+                        {
+                            "name": "CLUSTER_WIDE",
+                            "value": f"{args.cluster_wide}",
+                        },
+                        {
+                            "name": "OPERATOR_IMAGE",
+                            "value": f"{dev_config.repo_url}/{dev_config.operator_image}:{args.tag}",
+                        },
+                        {
+                            "name": "TEST_NAMESPACE",
+                            "value": dev_config.namespace,
+                        },
+                        {
+                            "name": "VERSION_UPGRADE_HOOK_IMAGE",
+                            # TODO: this needs to come from somewhere else
+                            "value": "quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2",
```

Comment:
If this doesn't need to be specified as an environment variable, maybe the VERSION_UPGRADE_HOOK_IMAGE could just live in some yaml/ as some constant. Or did you have something else in mind?

Reply:
when we need to make changes, it won't always be `1.0.2` it will be the `version_id` of an individual evergreen patch, or whatever tag we want if building locally. For now it isn't undergoing any changes, but this will need to be configurable.

Reply:
That makes sense, thank you for explaining!